deploy/dynamo/sdk/src/dynamo/sdk/cli/serving.py

```diff
@@ -127,6 +128,29 @@ def _get_server_socket(
 _DYNAMO_WORKER_SCRIPT = "dynamo.sdk.cli.serve_dynamo"
 
 
+def maybe_get_slurm_mpirun_command(service_name, args):
+    # HACK: WAR issues with running MPI_Spawn in TRTLLM internals
+    #       while in a Slurm environment.
+    # FIXME: Find a better solution for running TRTLLM workers with `dynamo serve`
+    is_slurm = os.environ.get("SLURM_NODELIST") is not None
+    if is_slurm and "TensorRTLLM" in service_name:
+        logger.info(
+            "Slurm environment detected: Pre-pending `mpirun` to TRTLLM workers."
```

Comment:
@tanmayv25 can you refresh my memory - is this only required in slurm environments, or everywhere? I assume TRTLLM always uses mpi comm spawn regardless of environment (at least for anything involving more than 1 gpu), so now I'm second guessing these slurm specific checks. Can you confirm the right conditions to add `mpirun`?

Reply:
I believe that this is more like a cluster specific configuration. I was able to run trtllm within dynamo serve without explicit mpirun on computelab and dlcluster. However, something about eos (and others) need the mpirun to launch.

Reply:
Do you have any idea what this missing link is? Do you have any ideas on what I can check to know when it would be needed?

Is it pyxis/enroot (`srun --container-image <image> ...`) vs. (`srun bash && docker run <image> ...`)?